### DISCUSSION TITLE
IR requiring KSP

### DISCUSSION THREAD

**Discussion by matejdro** [ZacSweers/MoshiX#498] [Q&A] [answered]
New version of the IR plugin now requires KSP.

Doesn't that kinda defeats its purpose? If we need to apply the KSP, we might as well just apply regular moshi through KSP. Or am I missing something?

Thanks.

**Comment by ZacSweers** [comment, marked as answer]
It doesn't require it, as the docs say you can disable it if you don't need proguard rule gen.

I'm hoping KSP will allow running KSP in parallel to kotlinc. You can also disable it in debug mode and only enable it for your release builds.

On top of all that, IR is still somewhat faster (no generating extra source files that need to be compiled) and allows doing more advanced things in generated adapters than we can't with Moshi-KSP (error aggregation, avoiding reflection with defaults constructors, etc).
